pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -689,11 +782,16 @@ public boolean isClosed() {
 
     @Override
     public synchronized void updateServiceUrl(String serviceUrl) throws PulsarClientException {
-        log.info("Updating service URL to {}", serviceUrl);
+        try {
+            log.info("Updating service URL to {}", serviceUrl);
 
-        conf.setServiceUrl(serviceUrl);
-        lookup.updateServiceUrl(serviceUrl);
-        cnxPool.closeAllConnections();
+            conf.setServiceUrl(serviceUrl);
+            lookup.updateServiceUrl(serviceUrl);
+            cnxPool.closeAllConnections();
+        } catch (Throwable t) {
+            log.warn("Failed to update service URL.");
+            close();
```

Comment:
Looks good, I have some questions
We only need to ensure that when closing all connections, we can ensure that all connections are called close, and that close failure of one connection will not cause other connections to not call close.
Therefore, updating the serviceUrl fails, why close the entire pulsarClient?

Reply:
@315157973  I didn't get you. Do you mean that failing to update the service URL will not stop the existing client and continue to work with the older service URL? If that is the case, then I can revert the changes to this method.

Reply:
yes

Reply:
Thanks, @315157973. I have reverted this change. can you help me with the failing tests? I have no idea why are they failing here.